src/hooks/video/agora/index.ts

```diff
@@ -0,0 +1,233 @@
+  UseAgoraCameraReturn,
+  UseAgoraRemotesProps,
+  UseAgoraRemotesReturn,
+  UseAgoraScreenShareProps,
+  UseAgoraScreenShareReturn,
+} from "types/agora";
+
+export const useAgoraRemotes: ReactHook<
+  UseAgoraRemotesProps,
+  UseAgoraRemotesReturn
+> = ({ client }) => {
+  const [remoteUsers, setRemoteUsers] = useState<IAgoraRTCRemoteUser[]>([]);
+
+  const updateRemoteUsers = useCallback(() => {
+    if (!client) return;
+
+    setRemoteUsers(() => Array.from(client.remoteUsers));
+  }, [client]);
+
+  const handleUserPublished = useCallback(
+    async (user: IAgoraRTCRemoteUser, mediaType: "audio" | "video") => {
+      if (!client) return;
+
+      await client.subscribe(user, mediaType);
+      updateRemoteUsers();
+    },
+    [client, updateRemoteUsers]
+  );
+
+  useEffect(() => {
+    if (!client) return;
+
+    updateRemoteUsers();
+
+    client.on("user-published", handleUserPublished);
+    client.on("user-unpublished", updateRemoteUsers);
+    client.on("user-joined", updateRemoteUsers);
+    client.on("user-left", updateRemoteUsers);
+    client.join(AGORA_APP_ID || "", AGORA_CHANNEL || "", AGORA_TOKEN || null);
+
+    return () => {
+      client.off("user-published", handleUserPublished);
+      client.off("user-unpublished", updateRemoteUsers);
+      client.off("user-joined", updateRemoteUsers);
+      client.off("user-left", updateRemoteUsers);
+      client.leave();
+    };
+  }, [client, handleUserPublished, updateRemoteUsers]);
+
+  return remoteUsers;
+};
+
+export const useAgoraScreenShare: ReactHook<
+  UseAgoraScreenShareProps,
+  UseAgoraScreenShareReturn
+> = ({ client }) => {
+  const { userId } = useUser();
+  const venueId = useVenueId();
+
+  const [localScreenTrack, setLocalScreenTrack] = useState<ILocalVideoTrack>();
+  const [localAudioTrack, setLocalAudioTrack] = useState<ILocalAudioTrack>();
+
+  const shareScreen = useCallback(async () => {
+    if (!client) return;
+
+    const screenTrack = await AgoraRTC.createScreenVideoTrack({}, "auto");
+
+    if (Array.isArray(screenTrack)) {
+      const [screenVideoTrack, screenAudioTrack] = screenTrack;
+      setLocalScreenTrack(screenVideoTrack);
+      setLocalAudioTrack(screenAudioTrack);
+      await client.publish(screenVideoTrack);
+      await client.publish(screenAudioTrack);
+      return;
+    }
+
+    setLocalScreenTrack(screenTrack);
+    await client.publish(screenTrack);
+  }, [client]);
+
+  const stopShare = useCallback(async () => {
+    localScreenTrack?.stop();
+    localScreenTrack?.close();
+
+    localAudioTrack?.stop();
+    localAudioTrack?.close();
+
+    await client?.unpublish();
+
+    setLocalScreenTrack(undefined);
+    setLocalAudioTrack(undefined);
+  }, [client, localAudioTrack, localScreenTrack]);
+
+  const joinChannel = async () => {
+    if (!client || !venueId || !userId) return;
+
+    const screenClientUid = await client?.join(
+      AGORA_APP_ID || "",
+      AGORA_CHANNEL || "",
+      AGORA_TOKEN || null
+    );
```

Comment:
Calling out usages of `AGORA_APP_ID`, `AGORA_CHANNEL`, and `AGORA_TOKEN` that are likely wrong and almost certainly will need refactoring.